tests/test_edflow.py

```diff
@@ -0,0 +1,221 @@
+        config["iterator"] = "tests." + fullname(Iterator1)
+        config["dataset"] = "tests." + fullname(Dataset)
+        config["batch_size"] = 16
+        config["num_steps"] = 100
+        import yaml
+
+        with open(os.path.join(tmpdir, "config.yaml"), "w") as outfile:
+            yaml.dump(config, outfile, default_flow_style=False)
+        import shutil
+
+        shutil.copytree(os.path.split(__file__)[0], os.path.join(tmpdir, "tests"))
+        command = [
+            "edflow",
+            "-e",
+            "config.yaml",
+            "-c",
+            os.path.join(
+                "logs", "trained_model", "train", "checkpoints", "model.ckpt-0"
+            ),
+            "-b",
+            "config.yaml",
+            "-n",
+            "test_inference",
+        ]
+        run_edflow_cmdline(command, cwd=tmpdir)
+
+        # check if correct folder was created
+        log_dirs = os.listdir(os.path.join(tmpdir, "logs"))
+        assert any(list(filter(lambda x: "test_inference" in x, log_dirs)))
+
+    def test_2(self, tmpdir):
+        """
+        Tests evaluation with providing a checkpoint and writing evaluation into the project folder.
+        This should load the checkpoint logs/trained_model/train/checkpoints/model.ckpt-0
+
+        effectively runs
+            edflow -e config.yaml -b config.yaml -c logs/trained_model/train/checkpoints/model.ckpt-0
+            -p logs/trained_model -n test_inference
+
+        and then checks if an evaluation folder "test_inference" was created in logs/trained_model/eval
+        -------
+        """
+        self.setup_tmpdir(tmpdir)
+        config = dict()
+        config["model"] = "tests." + fullname(Model)
+        config["iterator"] = "tests." + fullname(Iterator1)
+        config["dataset"] = "tests." + fullname(Dataset)
+        config["batch_size"] = 16
+        config["num_steps"] = 100
+        import yaml
+
+        with open(os.path.join(tmpdir, "config.yaml"), "w") as outfile:
+            yaml.dump(config, outfile, default_flow_style=False)
+        import shutil
+
+        shutil.copytree(os.path.split(__file__)[0], os.path.join(tmpdir, "tests"))
+        command = [
+            "edflow",
+            "-e",
+            "config.yaml",
+            "-c",
+            os.path.join(
+                "logs", "trained_model", "train", "checkpoints", "model.ckpt-0"
+            ),
+            "-b",
+            "config.yaml",
+            "-p",
+            os.path.join("logs", "trained_model"),
+            "-n",
+            "test_inference",
+        ]
+        run_edflow_cmdline(command, cwd=tmpdir)
+
+        # check if correct folder was created
+        eval_dirs = os.listdir(os.path.join(tmpdir, "logs", "trained_model", "eval"))
+        assert any(list(filter(lambda x: "test_inference" in x, eval_dirs)))
+
+    def test_3(self, tmpdir):
+        """
+        Tests evaluation without providing a checkpoint. This should NOT load any checkpoint.
```

Comment:
This should not load any checkpoint? I must say I'd prefer this to just load the latest checkpoint, no? I guess this is related to https://github.com/pesser/edflow/pull/62#discussion_r292508828

Reply:
Well that is the question. I think it would be most intuitive to do the following:

* `edflow -c ` ==> load checkpoint, evaluate into output folder
* `edflow -c ... -p` ==> load checkpoint AND evaluate into project folder
* `edflow -p` ==> load latest checkpoint AND evaluate into project folder.

This is basically what test1 ... test3 cover. 

In #62, test 3 goes through because the latest checkpoint is loaded eventually through the WaitForCheckpoint hook.

That having said, I think https://github.com/pesser/edflow/pull/62#discussion_r292508828 would comply with my desired high-level interface.